src/EFCore.PG/Extensions/NpgsqlBinaryStringExtensions.cs

```diff
@@ -0,0 +1,119 @@
+// THE POSSIBILITY OF SUCH DAMAGE.
+//
+// THE NPGSQL DEVELOPMENT TEAM SPECIFICALLY DISCLAIMS ANY WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE. THE SOFTWARE PROVIDED HEREUNDER IS
+// ON AN "AS IS" BASIS, AND THE NPGSQL DEVELOPMENT TEAM HAS NO OBLIGATIONS
+// TO PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
+
+#endregion
+
+using System;
+using System.Runtime.CompilerServices;
+using JetBrains.Annotations;
+
+// ReSharper disable once CheckNamespace
+namespace Microsoft.EntityFrameworkCore
+{
+    // ReSharper disable UnusedParameter.Global
+    /// <summary>
+    /// Provides extension methods for <see cref="DbFunctions"/> supporting PostgreSQL binary string functions and operators.
+    /// </summary>
+    public static class NpgsqlBinaryStringExtensions
+    {
+        // ReSharper disable once InconsistentNaming
+        /// <summary>
+        /// Computes the MD5 hash for the input data.
+        /// </summary>
+        /// <param name="_">The <see cref="DbFunctions"/> instance.</param>
+        /// <param name="data">The data to hash.</param>
+        /// <returns>
+        /// The MD5 hash of the input data.
+        /// </returns>
+        /// <exception cref="NotSupportedException">{method} is only intended for use via SQL translation as part of an EF Core LINQ query.</exception>
+        [NotNull]
+        public static string MD5([CanBeNull] this DbFunctions _, [NotNull] string data) => throw ClientEvaluationNotSupportedException();
+
+        // ReSharper disable once InconsistentNaming
+        /// <summary>
+        /// Computes the SHA224 hash for the input data.
+        /// </summary>
+        /// <param name="_">The <see cref="DbFunctions"/> instance.</param>
+        /// <param name="data">The data to hash.</param>
+        /// <returns>
+        /// The SHA224 hash of the input data.
+        /// </returns>
+        /// <exception cref="NotSupportedException">{method} is only intended for use via SQL translation as part of an EF Core LINQ query.</exception>
+        [NotNull]
+        public static byte[] SHA224([CanBeNull] this DbFunctions _, [NotNull] byte[] data) => throw ClientEvaluationNotSupportedException();
+
+        // ReSharper disable once InconsistentNaming
+        /// <summary>
+        /// Computes the SHA256 hash for the input data.
+        /// </summary>
+        /// <param name="_">The <see cref="DbFunctions"/> instance.</param>
+        /// <param name="data">The data to hash.</param>
+        /// <returns>
+        /// The SHA256 hash of the input data.
+        /// </returns>
+        /// <exception cref="NotSupportedException">{method} is only intended for use via SQL translation as part of an EF Core LINQ query.</exception>
+        [NotNull]
+        public static byte[] SHA256([CanBeNull] this DbFunctions _, [NotNull] byte[] data) => throw ClientEvaluationNotSupportedException();
```

Comment:
How about translating [the built-in SHA256](https://docs.microsoft.com/en-gb/dotnet/api/system.security.cryptography.sha256?view=netcore-2.1) class instead of adding this? Note that this class supports the concept of different implementations, we may want to understand if we want to translate all of them (probably yes) or not.

This is probably relevant for at least some of the other hashing functions you're translating here.